wcfsetup/install/files/ts/WoltLabSuite/Core/Media/Manager/Editor.ts

```diff
@@ -0,0 +1,395 @@
+      } else {
+        const blobUploadData = this._uploadData as PasteFromClipboard;
+        this._uploadId = this._upload.uploadBlob(blobUploadData.blob);
+      }
+
+      this._uploadData = null;
+    }
+  }
+
+  /**
+   * Handles pasting and dragging and dropping files into the editor.
+   */
+  protected _editorUpload(data: OnDropPayload): void {
+    this._uploadData = data;
+
+    UiDialog.open(this);
+  }
+
+  /**
+   * Returns the id of the insert dialog based on the media files to be inserted.
+   */
+  protected _getInsertDialogId(): string {
+    let dialogId = "mediaInsert";
+
+    this._mediaToInsert.forEach((media, mediaId) => {
+      dialogId += `-${mediaId}`;
+    });
+
+    return dialogId;
+  }
+
+  /**
+   * Returns the supported thumbnail sizes (excluding `original`) for all media images to be inserted.
+   */
+  protected _getThumbnailSizes(): string[] {
+    return ["small", "medium", "large"]
+      .map((size) => {
+        const supportSize = Array.from(this._mediaToInsert).every(([_mediaId, media]) => {
+          return media[size + "ThumbnailType"] !== null;
+        });
+
+        if (supportSize) {
+          return size;
+        }
+
+        return null;
+      })
+      .filter((s) => s !== null) as string[];
+  }
+
+  /**
+   * Inserts media files into the editor.
+   */
+  protected _insertMedia(event?: Event | null, thumbnailSize?: string, closeEditor = false): void {
+    if (closeEditor === undefined) closeEditor = true;
+
+    const insertType = "separate";
+
+    // update insert options with selected values if method is called by clicking on 'insert' button
+    // in dialog
+    if (event) {
+      UiDialog.close(this._getInsertDialogId());
+
+      const dialogContent = (event.currentTarget as HTMLElement).closest(".dialogContent")!;
+      const thumbnailSizeSelect = dialogContent.querySelector("select[name=thumbnailSize]") as HTMLSelectElement;
+      thumbnailSize = thumbnailSizeSelect.value;
+    }
+
+    if (this._options.callbackInsert !== null) {
+      this._options.callbackInsert(this._mediaToInsert, insertType, thumbnailSize!);
+    } else {
+      if (insertType === "separate") {
+        this._options.editor!.buffer.set();
+
+        this._mediaToInsert.forEach((media) => this._insertMediaItem(thumbnailSize!, media));
+      } else {
+        this._insertMediaGallery();
+      }
+    }
+
+    if (this._mediaToInsertByClipboard) {
+      Clipboard.unmark("com.woltlab.wcf.media", Array.from(this._mediaToInsert.keys()));
+    }
+
+    this._mediaToInsert = new Map<number, Media>();
+    this._mediaToInsertByClipboard = false;
+
+    // close manager dialog
+    if (closeEditor) {
+      UiDialog.close(this);
+    }
+  }
+
+  /**
+   * Inserts a series of uploaded images into the editor using a slider.
+   */
+  protected _insertMediaGallery(): void {
+    const mediaIds: number[] = [];
+    this._mediaToInsert.forEach(function (media) {
+      mediaIds.push(media.mediaID);
+    });
```

Comment:
```suggestion
    const mediaIds = Array.from(this._mediaToInsert.values()).map((media) => media.mediaID);
```

(And: Aren't the keys of that map already the IDs?)